tests/unit/via/services/youtube_transcript_test.py

```diff
@@ -0,0 +1,300 @@
+import json
+from io import BytesIO
+from json import JSONDecodeError
+from unittest.mock import sentinel
+from xml.etree import ElementTree
+
+import pytest
+from h_matchers import Any
+from requests import Response
+
+from tests.factories import TranscriptInfoFactory
+from via.exceptions import UnhandledUpstreamException
+from via.services.youtube_transcript import (
+    TranscriptInfo,
+    YouTubeTranscriptService,
+    factory,
+)
+
+
+class TestTranscriptInfo:
+    @pytest.mark.parametrize(
+        "transcript_info,expected_id",
+        [
+            (
+                TranscriptInfoFactory(),
+                "en-us..RW5nbGlzaCAoVW5pdGVkIFN0YXRlcyk=",
+            ),
+            (
+                TranscriptInfoFactory(autogenerated=True),
+                "en-us.a.RW5nbGlzaCAoVW5pdGVkIFN0YXRlcyk=",
```

Comment:
And ideas for auto-generated transcripts look like this: `"{language_code}.a.{name}"`.